lib/transaction/transaction_pool.go

```diff
@@ -59,52 +56,30 @@ func (tp *TransactionPool) Remove(hashes ...string) {
 	tp.Lock()
 	defer tp.Unlock()
 
-	indices := map[int]int{}
-	var max int
 	for _, hash := range hashes {
-		index, found := common.InStringArray(tp.Hashes, hash)
-		if !found {
-			continue
-		}
-		indices[index] = 1
-		if index > max {
-			max = index
-		}
-
-		if tx, found := tp.Get(hash); found {
+		if tx, found := tp.Pool[hash]; found {
 			delete(tp.Sources, tx.Source())
+			delete(tp.Pool, hash)
 		}
 	}
-
-	var newHashes []string
-	for i, hash := range tp.Hashes {
-		if i > max {
-			newHashes = append(newHashes, hash)
-			continue
-		}
-
-		if _, found := indices[i]; !found {
-			newHashes = append(newHashes, hash)
-			continue
-		}
-
-		delete(tp.Pool, hash)
-	}
-
-	tp.Hashes = newHashes
-
-	return
 }
 
 func (tp *TransactionPool) AvailableTransactions(transactionLimit int) []string {
+	if transactionLimit < 1 {
+		panic("Invalid transaction limit specified")
```

Comment:
That should never be triggered (not handled).
If that panic is triggered, it's because there is a bug in the code.

Reply:
I prefer panic like assertion in C language to ensure that should not happen.
However, to adopt this strategy we need a sufficient test for sure.
I am worried that propagation of system failure caused by this kind of fault without noticed.

If we allow panic(or regardless of panic), we need to prepare the middleware.

Reply:
@kfangw : That's exactly the way it's intended here (C-style `assert`).

You don't usually test `assert` because they are a logical failure, outside of the realm of the function's responsibility. In other words, no one should ever rely on a function's `assert`.

In this case, the function should really accept an unsigned number, as passing a negative transactionLimit is just wrong.

Reply:
How about casting uint instead of int here and get uint type?
https://github.com/bosnet/sebak/blob/fdd28f3de34f322be685ba38f1308ea2134c4499/lib/node/runner/node_runner.go#L475
@Geod24